docs/toolhive/concepts/tool-optimization.mdx

```diff
@@ -0,0 +1,392 @@
+
+This means when you override a tool name, you must use the new name in your
+filter list, not the original name.
+
+### Pattern: Secure subset with clear names
+
+Start by overriding technical names to be more intuitive, then filter to only
+safe operations.
+
+```json
+{
+  "toolsOverride": {
+    "exec_raw_sql": {
+      "name": "run_database_query",
+      "description": "Execute read-only SQL queries against the staging database"
+    },
+    "write_table": {
+      "name": "update_database",
+      "description": "Modify staging database tables (use with caution)"
+    }
+  }
+}
+```
+
+Then filter using the new names:
+
+```bash
+thv run --tools-override overrides.json --tools run_database_query my-db-server
+```
+
+**Why this works**: The combination gives you both clarity and safety. The AI
+sees `run_database_query` with a helpful description that makes the purpose
+obvious, and write operations aren't available at all. This pattern is excellent
+for development and staging environments where you want to prevent accidental
+data modifications.
+
+### Pattern: Environment-specific configurations
+
+Different environments need different tool access. In development, you might
+expose many tools for flexibility. In production, filter to essential tools
+only.
+
+Your development configuration could expose all tools with friendly names
+through overrides. Your production configuration uses the same overrides for
+consistency but adds strict filtering to expose only read and monitoring tools,
+blocking any write or deployment operations.
+
+**Why this works**: The same override configuration ensures consistent tool
+names across environments, making it easier to write documentation and train
+team members. The filtering layer adds environment-appropriate safety without
+requiring different tool naming schemes. This prevents production accidents
+while maintaining development flexibility.
+
+### Pattern: Multi-server aggregation
+
+When using Virtual MCP Server to combine multiple MCP servers, overrides prevent
+conflicts and improve clarity:
+
+```json
+{
+  "toolsOverride": {
+    "search": {
+      "name": "github_search",
+      "description": "Search GitHub repositories and code"
+    }
+  }
+}
+```
+
+You can override the `search` tool from different servers to `github_search`,
+`jira_search`, and `confluence_search`. Then filter each server to its relevant
+tools, creating a clean, conflict-free tool set.
+
+**Why this works**: Tool name prefixes make it impossible to accidentally call
+the wrong server's version of a tool. Filtering ensures you're only exposing the
+most relevant tools from each server, preventing context overload even when
+aggregating many services. The AI can confidently select tools knowing the
+prefix indicates the target system.
+
+## Making decisions about filtering and overrides
+
+When setting up an MCP server, consider these questions:
+
+### Do you need filtering?
```

Comment:
There is a section about when to use filtering. Would this be too much repetition?